dts/dts-tests.robot

```diff
@@ -0,0 +1,220 @@
+Library             RequestsLibrary
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../variables.robot
+
+Suite Setup         Run Keywords
+...                     Prepare Test Suite    AND
+...                     Skip If    not ${DTS_SUPPORT}
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Variables ***
+${DTS_CHECKPOINT}=      Enter an option:
+
+
+*** Test Cases ***
+PC Engines DPP initial deployment (legacy -> UEFI) - no credentials
+    [Documentation]    Verify DPP (coreboot + UEFI) initial deployment logic on
+    ...    PC Engines. We start from legacy firmware and insert no DPP keys. No
+    ...    firmware shall be offered, but we expect link to shop.
+
+    Power On And Enter DTS Shell
+
+    Execute Command In Terminal
+    ...    export BOARD_VENDOR="PC Engines" SYSTEM_MODEL="APU2" BOARD_MODEL="APU2" SYSTEM_VENDOR="PC Engines"
+    Execute Command In Terminal    export BIOS_VERSION="v4.19.0.1"
+    Write Into Terminal    dts-boot
+
+    ${out}=    Read From Terminal Until    ${DTS_CHECKPOINT}
+    Write Into Terminal    2
+    Log    ${out}
+
+    ${out}=    Read From Terminal Until    Enter an option:
+    Should Contain
+    ...    ${out}
+    ...    DPP version (coreboot + UEFI) available, if you are interested, please visit https://shop.3mdeb.com/product-category/dasharo-entry-subscription/
+    Should Contain
+    ...    ${out}
+    ...    DPP version (coreboot + SeaBIOS) available, if you are interested, please visit https://shop.3mdeb.com/product-category/dasharo-entry-subscription/
+    Log    ${out}
+    Write Into Terminal    b
+
+PC Engines DPP initial deployment (legacy -> UEFI)
+    [Documentation]    Verify DPP (coreboot + UEFI) initial deployment logic on
+    ...    PC Engines. We start from legacy firmware and insert correct DPP keys for
+    ...    UEFI variant.
+
+    Power On And Enter DTS Shell
+
+    Execute Command In Terminal
+    ...    export BOARD_VENDOR="PC Engines" SYSTEM_MODEL="APU2" BOARD_MODEL="APU2" SYSTEM_VENDOR="PC Engines"
+    Execute Command In Terminal    export BIOS_VERSION="v4.19.0.1" DPP_TYPE="UEFI"
+    Write Into Terminal    dts-boot
+
+    ${out}=    Read From Terminal Until    ${DTS_CHECKPOINT}
+    Write Into Terminal    2
+    Log    ${out}
+
+    ${out}=    Read From Terminal Until    Enter an option:
+    Should Contain    ${out}    d) DPP version (coreboot + UEFI)
+    Log    ${out}
+    Write Into Terminal    d
+
+    ${out}=    Read From Terminal Until    Does it match your actual specification? (Y|n)
+    Write Into Terminal    Y
+    Log    ${out}
+
+    ${out}=    Read From Terminal Until    Do you want to install Dasharo firmware on your hardware? (Y|n)
+    Write Into Terminal    Y
+    Log    ${out}
+
+    ${out}=    Read From Terminal Until    Rebooting
```

Comment:
Will it get here? In DTS scripts, you are communicating with hardware not only via `dmidecode` in `dts-environment.sh` but in many other ways, including `flashrom` read/write commands, and all of them will fail on QEMU. So, will the test get here if you will launch it on QEMU?

Reply:
And here is a confirmation. I have been trying to go through your second test (which is `PC Engines DPP initial deployment (legacy -> SeaBIOS)`), and it did not pass:

```bash
==============================================================================
Dts-Tests                                                                     
==============================================================================
PC Engines DPP initial deployment (legacy -> UEFI) :: Verify DPP (... | FAIL |
No match found for 'Rebooting' in 6 minutes 30 seconds. Output:

Backing up BIOS firmware and store it locally...
Remember that firmware is also backed up in HCL report.
Failed to read BIOS firmware backup : (1)
Press any key to continue.
------------------------------------------------------------------------------
```

As you can see it has failed on one of the `flashrom` commands ([here](https://github.com/Dasharo/dts-scripts/blob/5ebb3d910764a4589c6473a577c822967b3b0d19/scripts/dasharo-deploy#L207)), because `flashrom` does not work with QEMU this could be considered as false negative.

Reply:
Shouldn't `$FLASHROM` be set to https://github.com/Dasharo/dts-scripts/blob/main/tests/flashrom-mock.sh? That's how these tests work, by either setting variables before (so `dmidecode` isn't used) or by using mock/fake version of tool

Reply:
`FLASHROM="${FLASHROM:-flashrom}"` this will set `FLASHROM` to `flashrom` only if variable isn't already set. Maybe you missed some configuration/preparation step?

Reply:
Exactly. See: https://github.com/Dasharo/dts-scripts/tree/main/tests#running-manually and https://github.com/Dasharo/dts-scripts/blob/main/tests/dts-boot

Reply:
It should, but do we have all needed mock/fake versions now? What about handling calls to tools, which read hardware inf., e.g. [during configuration](https://github.com/Dasharo/dts-scripts/blob/5ebb3d910764a4589c6473a577c822967b3b0d19/include/dts-functions.sh#L371)

Reply:
We have what can be seen. So flashrom end ectool, and that was enough to run the scenarios shown in the first comment at the time of writing this.

We never know if we will not execute these (or other) tests any longer.

The linked code is wrong, it should use variable not command directly. Clearly it will cause troubles in qemu.

Reply:
:writing_hand:

Reply:
We could prepend `PATH`  so it contains our `fake` tools, just rename to correct names e.g. `flashrom-mock.sh` to `flashrom`. Linux should use tool from first available directory in `PATH`